sycl/include/sycl/vector_preview.hpp

```diff
@@ -363,18 +359,28 @@ template <typename T>
 using vec_data_t = typename detail::vec_helper<T>::RetType;
 
 ///////////////////////// class sycl::vec /////////////////////////
-/// Provides a cross-patform vector class template that works efficiently on
-/// SYCL devices as well as in host C++ code.
-///
-/// \ingroup sycl_api
+// Provides a cross-patform vector class template that works efficiently on
+// SYCL devices as well as in host C++ code.
 template <typename Type, int NumElements>
 class vec : public detail::vec_arith<Type, NumElements> {
   using DataT = Type;
 
+  static constexpr size_t AdjustedNum = (NumElements == 3) ? 4 : NumElements;
```

Comment:
1.  I rearranged this with the intention of keeping the widely used type traits/aliases/constexpr variables at the beginning of the class `sycl::vec`.  This would hopefully make `sycl::vec` more readable.
2. Also, in my followup PRs (split-ups of https://github.com/intel/llvm/pull/13270), I will remove/inline all type traits/aliases after `vector_t`. So, rearranging `AdjustedNum` now would make the diff cleaner later.

Reply:
I actually mean why do we convert AdjustedNum from 3 to 4.

Reply:
Ahh... I see. https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#memory-layout-and-alignment

> The exception to this is when the number of element is three in which case the SYCL vec is aligned to the size of the element type in bytes multiplied by four:

It's required by the SPEC to align `sycl::vec` with OpenCL vectors.

Reply:
Can you add a comment in the code pointing to this link?

Reply:
Sure. Added in https://github.com/intel/llvm/pull/13949/commits/d0f6e0cec0bfd648c390b7dc1143d510fdbdfa4e